### PR TITLE
feat(aws): init AWS CloudWatch

### PR DESCRIPTION
**Pull request by mlycore**
### What is changed and how it works?

- Add a workspace `audit/aws` for AWS CloudWatch

Further work needed.

### CODE REVIEW

File: pisa-proxy/audit/aws/src/aws.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                let resp = self
+                    .client
+                    .put_log_events()
+                    .set_sequence_token(Some(next.to_string()))
+                    .log_group_name(log_group_name.clone())
+                    .log_stream_name(log_stream_name)
+                    .log_events(e)
+                    .send()
+                    .await?;
+                println!("aws resp {:?}", resp);
```
Comment (wbtlb) [inline pisa-proxy/audit/aws/src/aws.rs:76]: replace` println!` to `tracing`

File: pisa-proxy/audit/aws/src/aws.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    pub async fn send(&self, input: CloudWatchLog) -> Result<(), aws_sdk_cloudwatchlogs::Error> {
+        let log_group_name = input.log_group_name.clone();
+        let log_stream_name = input.log_stream_name.clone();
+        let message = input.message.clone();
+
+        let streams =
+            self.client.describe_log_streams().log_group_name(input.log_group_name).send().await?;
+        for s in streams.log_streams().unwrap() {
+            if s.log_stream_name().unwrap() == log_stream_name.clone() {
+                let next = s.upload_sequence_token().unwrap();
```
Comment (wbtlb) [inline pisa-proxy/audit/aws/src/aws.rs:60]: consider report unwrap by `?`.

File: pisa-proxy/audit/aws/src/aws.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            log_stream_name: stream,
+            timestamp: now.timestamp_millis(),
+            message,
+        }
+    }
+}
+
+pub struct CloudWatchSinker {
+    pub client: Client,
+    // pub region: String,
```
Comment (wbtlb) [inline pisa-proxy/audit/aws/src/aws.rs:40]: remove unused code.

Comment (mlycore) [reply pisa-proxy/audit/aws/src/aws.rs:40]: fixed

File: pisa-proxy/audit/aws/src/aws.rs (hunk 1 above)
Comment (mlycore) [reply pisa-proxy/audit/aws/src/aws.rs:76]: > replace` println!` to `tracing`

fixed

File: pisa-proxy/audit/aws/src/aws.rs (hunk 2 above)
Comment (mlycore) [reply pisa-proxy/audit/aws/src/aws.rs:61]: fixed

**Review by wbtlb — APPROVED** [review]

**Pull request merged**